projects/worker-ccai-pdfextract/worker-ccai-pdfextract.js

```diff
@@ -0,0 +1,172 @@
+ * @property {Number} score confidence score (0.0-1.0)
+ */
+/**
+ * Parse keywords in feature response
+ * 
+ * @param {*} response JSON response from Content and Commerce AI service
+ * @returns {Entity[]} Color features returned by the service
+ */
+function parseEntities(response) {
+    const entities = [];
+    const resp = parseResponse(response);
+    for (const features of resp) {
+        if (features.feature_name === CONTENT_ID) {
+            for (const feature of features.feature_value) {
+                if (feature.feature_name === "labels") {
+                    const labels = feature.feature_value // array of feature values and names of labels
+                    for (const label of labels) {
+                        const valuesObj = {};
+                        label.feature_value.forEach(f => {
+                            valuesObj[f.feature_name] = f.feature_value;
+                        });
+                        entities.push(Object.assign({
+                            name: label.feature_name,
+                        }, valuesObj));
+                    }
+                }
+            }
+        }
+    }
+    return entities;
+}
+
+/**
+ * Sort entity scores, high to low confidence
+ * 
+ * @param {Entity[]} entities entity features
+ * @return {Entity[]} Color features sorted by percentage (high to low)
+ */
+function sortScores(entities) {
+    entities.sort((a, b) => b.score - a.score);
+}
+
+exports.main = worker(async (source, rendition, params) => {
+    // Acquire end point and analyzer
+    const analyzer_id = rendition.instructions.ANALYZER_ID || DEFAULT_ANALYZER_ID;
+    const endpoint = rendition.instructions.CCAI_ENDPOINT || DEFAULT_CCAI_ENDPOINT;
+    console.log("Using analyzer:", analyzer_id);
+    console.log("Using endpoint:", endpoint);
+
+    // Make sure that the source file is not empty
+    const stats = await fs.stat(source.path);
+    if (stats.size === 0) {
+        throw new SourceCorruptError('source file is empty');
+    }
+
+    // Build parameters to send to Sensei service
+    const parameters = {
+        "application-id": "1234",                             
+        "content-type": "file",                                         
+        "encoding": "pdf",
+        "threshold": 0.01,
+        "top-N": DEFAULT_MAX_RESULTS,                                   
+        "data": [                                
+          {                                   
+            "content-id": CONTENT_ID,                 
+            "content": "file"         
+          }                  
+        ]
+    };
+
+    // Build form to post
+    const formData = new FormData();
+    formData.append('file', createReadStream(source.path));
+    formData.append('contentAnalyzerRequests', JSON.stringify({
+        enable_diagnostics: true,
+        requests: [{
+            analyzer_id,
+            parameters,
+            "content_id": "test123",
```

Comment:
Is this a debug line?

Reply:
its not required according to the api:
```
The unique ID for the data element thats returned in the response. If this is not passed an auto-generated ID is assigned.
```

so i'll go ahead and remove it